std/assembly/rt/tlsf.ts

```diff
@@ -475,7 +475,14 @@ export function maybeInitialize(): Root {
         SETHEAD(root, fl, sl, null);
       }
     }
-    addMemory(root, (rootOffset + ROOT_SIZE + AL_MASK) & ~AL_MASK, memory.size() << 16);
+    if (ASC_LOW_MEMORY_LIMIT > 0) {
+      const start = (rootOffset + ROOT_SIZE + AL_MASK) & ~AL_MASK;
+      const end = (<usize>ASC_LOW_MEMORY_LIMIT + AL_MASK) & ~AL_MASK;
+      if (start > end) ERROR("Low memory limit exceeded by TLSF bookkeeping");
+      else if (start < end) addMemory(root, start, end);
+    } else {
+      addMemory(root, (rootOffset + ROOT_SIZE + AL_MASK) & ~AL_MASK, memory.size() << 16);
+    }
```

Comment:
Simplify?
```ts
    const start = (rootOffset + ROOT_SIZE + AL_MASK) & ~AL_MASK;
    let end: usize;
    if (ASC_LOW_MEMORY_LIMIT > 0) {
      end = (<usize>ASC_LOW_MEMORY_LIMIT + AL_MASK) & ~AL_MASK;
      if (start > end) ERROR("Low memory limit exceeded by TLSF bookkeeping");
    } else {
      end = memory.size() << 16;
    }
    addMemory(root, start, end);
```

Reply:
Looks like `let end: usize;` would break precompute there.

Reply:
Ah, right. I forgot what it is conditional compilation)

Reply:
But now that you mentioned it, it seems `start` does not precompute anyway due to depending on `__heap_base`. Hmm...